apps/prairielearn/src/pages/error/error.js

```diff
@@ -5,6 +5,72 @@ var jsonStringifySafe = require('json-stringify-safe');
 
 const { logger } = require('@prairielearn/logger');
 
+/**
+ * @param {string} stack
+ * @param {number} depth
+ * @returns {string}
+ */
+function indentString(stack, depth) {
+  if (depth === 0) return stack;
+
+  const indent = '    '.repeat(depth);
+  return stack
+    .split('\n')
+    .map((line) => (indent + line).trimEnd())
+    .join('\n');
+}
+
+/**
+ * Recursively formats an error into a string. Correctly handles both the
+ * `.cause` property and `AggregateError` instances.
+ *
+ * @param {any} err
+ * @param {number} [depth]
+ * @param {string} [prefix]
+ * @returns {string}
+ */
+function formatErrorStack(err, depth = 0, prefix = '') {
```

Comment:
Maybe this belongs in `@prairielearn/error` so we can use it in PrairieTest?

Reply:
Yep, it would be great to share this. PT currently has its own recursive rendering of errors: https://github.com/PrairieLearnInc/PrairieTest/blob/87b31fad317f99fdb946f94f24cc87c047ef1d1d/src/pages/error/error.html.ts#L141 but this version is better and it would be great to use a common library for this.

Reply:
Moved to `@prairielearn/error` in 5b5ec8c!

Reply:
Thanks!